crates/task-impls/src/vid.rs

```diff
@@ -309,7 +309,7 @@ where
                 // `self.cur_view` should be at least 1 since there is a view change before getting
                 // the `DAProposalRecv` event. Otherewise, the view number subtraction below will
                 // cause an overflow error.
-                if view < self.cur_view - 1 {
+                if *self.cur_view == 0 || view < self.cur_view - 1 {
```

Comment:
If we are going to use two conditions here, I think the first one should be `if *self.cur_view != 0`, and `||` should be `&&`.

Anyway, this is equivalent to my change in the [block payload PR](https://github.com/EspressoSystems/HotShot/pull/1968/files#diff-73ecf69c8525d26ad29ee60d0cfc62de3a8316ca47ab2946f0c64e42ee976590), as discusses on Zulip.

Reply:
I'll take the changes from your PR